msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplicationConfiguration.kt

```diff
@@ -0,0 +1,204 @@
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client
+
+import android.text.TextUtils
+import com.google.gson.annotations.SerializedName
+import com.microsoft.identity.client.configuration.AccountMode
+import com.microsoft.identity.client.exception.MsalClientException
+import com.microsoft.identity.common.java.authorities.CIAMAuthority
+import com.microsoft.identity.common.java.authorities.NativeAuthCIAMAuthority
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.logging.Logger
+import lombok.Getter
+import lombok.experimental.Accessors
+import java.io.Serializable
+
+/**
+ * Extends [PublicClientApplicationConfiguration] to add a few extra fields specific to Native Auth
+ * app implementation.
+ */
+@Accessors(prefix = ["m"])
+@Getter
+class NativeAuthPublicClientApplicationConfiguration :
+    PublicClientApplicationConfiguration(),
+    Serializable {
+    companion object {
+        private val TAG = NativeAuthPublicClientApplicationConfiguration::class.java.simpleName
+        private val VALID_CHALLENGE_TYPES = listOf("password", "oob", "redirect")
+    }
+
+    private object NativeAuthSerializedNames {
+        const val CHALLENGE_TYPES = "challenge_types"
+        const val USE_REAL_AUTHORITY = "useRealAuthority"
+        const val DC = "dc"
+    }
+
+    @SerializedName(NativeAuthSerializedNames.CHALLENGE_TYPES)
+    private var challengeTypes: List<String>? = null
+
+    @SerializedName(NativeAuthSerializedNames.USE_REAL_AUTHORITY)
+    var useRealAuthority: Boolean? = null
+
+    @SerializedName(NativeAuthSerializedNames.DC)
+    var dc: String? = null
+
+    fun getChallengeTypes(): List<String>? {
+        return challengeTypes
+    }
+
+    fun setChallengeTypes(challengeTypes: List<String>?) {
+        this.challengeTypes = challengeTypes
+    }
+
+    fun mergeConfiguration(config: NativeAuthPublicClientApplicationConfiguration) {
+        // Call super.mergeConfiguration to handle base configuration fields
+        super.mergeConfiguration(config)
+
+        // Native-auth specific account mode check. If dev passes MULTIPLE, we still want to set it so
+        // we can throw an exception to clarify that MULTIPLE account mode can't be used with Native
+        // Auth.
+        accountMode = if (config.accountMode != null) config.accountMode else accountMode
+
+        // Handle Native Auth specific fields
+        challengeTypes = if (config.challengeTypes == null) challengeTypes else config.challengeTypes
```

Comment:
since useRealAuthority and dc were added, we should probably add them to the mergeConfiguration method

Reply:
Addressed